projects/plugins/jetpack/extensions/shared/components/product-management-controls/index.js

```diff
@@ -0,0 +1,70 @@
+/**
+ * WordPress dependencies
+ */
+import { BlockControls } from '@wordpress/block-editor';
+import { useSelect } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import ProductManagementInspectorControl from './inspector-control';
+import ProductManagementToolbarControl from './toolbar-control';
+import InvalidProductWarning from './invalid-product-warning';
+import StripeConnectToolbarButton from '../stripe-connect-toolbar-button';
+import { store as membershipProductsStore } from '../../../store/membership-products';
+
+import './style.scss';
+
+export default function ProductManagementControls( {
+	allowCreateOneTimeInterval = true,
+	blockName,
+	selectedProductId = 0,
+	setSelectedProductId = () => {},
+} ) {
+	const products = useSelect(
+		select =>
+			select( membershipProductsStore ).getProducts( selectedProductId, setSelectedProductId ),
+		[]
+	);
+	const { connectUrl, isApiConnected, isSelectedProductInvalid, shouldUpgrade } = useSelect(
+		select => {
+			const { getConnectUrl, getShouldUpgrade, isApiStateConnected, isInvalidProduct } = select(
+				membershipProductsStore
+			);
+			return {
+				connectUrl: getConnectUrl(),
+				isApiConnected: isApiStateConnected(),
+				isSelectedProductInvalid: isInvalidProduct( selectedProductId ),
+				shouldUpgrade: getShouldUpgrade(),
+			};
+		}
+	);
+
+	if ( shouldUpgrade ) {
+		return null;
+	}
+
+	return (
+		<>
+			{ ! isApiConnected && !! connectUrl && (
+				<BlockControls group="block">
+					<StripeConnectToolbarButton blockName={ blockName } connectUrl={ connectUrl } />
+				</BlockControls>
+			) }
+			{ isApiConnected && (
+				<>
+					<ProductManagementInspectorControl
```

Comment:
I think we have a problem with the current approach. This approach doesn't play well with blocks that have their own InspectorControls.  We end up with more than one InspectorControls declaration. I don't think there is an easy solution to this problem, but maybe we could export each individual control type (ProductManagementInspectorControl and ProductManagementToolbarControl), and just have them return the PanelBody so that we can compose the controls as needed.